### PR TITLE
@anandaroop  => Remove calls to reloadable in favor of NPM package

### PR DESCRIPTION
**Pull request by damassi**
This PR removes existing calls to `reloadable` in favor of https://github.com/artsy/express-reloadable being placed at the root, and fixes an issue with `apps/react_example` not mounting client-side. Note that the `server.js` file is no longer needed and has been removed from each subapp.

### CODE REVIEW

File: desktop/assets/react_example.coffee [hunk 1]
```diff
@@ -3,7 +3,7 @@ require('backbone').$ = $
 routes =
   '''
   /react-example/
-  ''': require('../apps/react_example/client.js').default
+  ''': require('../apps/react_example/client.js').default()
```
Comment (anandaroop) [inline desktop/assets/react_example.coffee]: 👍

File: lib/setup.js [hunk 2]
```diff
@@ -202,6 +203,32 @@ export default function (app) {
   // Sets up mobile marketing signup modal
   app.use(marketingSignupModal)
 
+  // Setup hot-swap loader
+  if (NODE_ENV === 'development') {
+    const reloadAndMount = createReloadable(app, require)
```
Comment (anandaroop) [inline lib/setup.js]: Interesting, why does `require` have to be passed through like this?

File: desktop/apps/article2/server.js [hunk 3]
```diff
@@ -1,8 +0,0 @@
-import * as routes from './routes'
-import adminOnly from 'desktop/lib/admin_only.coffee'
-import express from 'express'
```
Comment (anandaroop) [inline desktop/apps/article2/server.js]: 🔪 🔪 🔪

File: lib/setup.js (hunk 2 above)
Comment (damassi) [inline lib/setup.js]: This is due to how modules work in Node, and scope; from the perspective of each library, each has its own `require` context and cache and when a file changes it needs to know where to look to find and remove it. In this sense, while `require` is available throughout the entirely of a Node app it's not exactly a global.

Comment (anandaroop) [inline lib/setup.js]: I see, thanks for the explanation.

**Pull request merged**